share/config.fish

```diff
@@ -66,17 +66,47 @@ if test -f $__fish_data_dir/__fish_build_paths.fish
     source $__fish_data_dir/__fish_build_paths.fish
 end
 
+# Compute the directories for vendor configuration.  We want to include
+# all of XDG_DATA_DIRS, as well as the __extra_* dirs defined above.
+set -l xdg_data_dirs
+set -l extra_mode
+if set -q XDG_DATA_DIRS
+    set --path xdg_data_dirs $XDG_DATA_DIRS
+    # Let XDG_DATA_DIRS take precedence because they may have user-specific overrides.
+    set extra_mode --append
+else
+    set xdg_data_dirs $__fish_data_dir
+    # The extra directories default to /usr/local/share/*
+    # We need to have them take precedence over $__fish_data_dir
+    set extra_mode --prepend
```

Comment:
Second thought --- I wonder if the additional complexity with append/prepend is necessary.
The intention was to fix the order to be like `/usr/local/share:/usr/share`.
At the moment this will use `${extra_completionsdir}:$__fish_datadir/vendor_completions.d`, both are in `/usr/local/share` by default. So why not swap them, and have the "extra" dirs always come last.

So far there has only been one directory called `vendor_completions.d` - the `extra_completionsdir` (that is created at installation time).

Reply:
Yes, that makes sense.